Mathlib/RingTheory/Spectrum/Prime/Homeomorph.lean

```diff
@@ -76,6 +76,46 @@ lemma PrimeSpectrum.isHomeomorph_comap_of_bijective {f : R →+* S} (hf : Functi
 
 open Algebra
 
+variable {k R K} in
+lemma IsPurelyInseparable.exists_pow_mem_range_tensorProduct [IsPurelyInseparable k K]
+    (x : R ⊗[k] K) : ∃ n > 0, x ^ n ∈ (algebraMap R (R ⊗[k] K)).range := by
+  wlog hR : Nontrivial R
+  · rw [not_nontrivial_iff_subsingleton] at hR
+    refine ⟨1, Nat.one_pos, 1, Subsingleton.elim _ _⟩
+  let q := ringExpChar k
+  obtain (hq|hq) := expChar_is_prime_or_one k q
+  · obtain ⟨n, hn, hr⟩ : ∃ n > 0, x ^ q ^ n ∈ (algebraMap R (R ⊗[k] K)).range := by
+      induction x with
+      | zero => exact ⟨1, Nat.zero_lt_one, 0, by simp [zero_pow_eq, hq.ne_zero]⟩
+      | add x y hx hy =>
+        obtain ⟨n, hn, hx⟩ := hx
+        obtain ⟨m, hm, hy⟩ := hy
+        refine ⟨n + m, by simp [hn], ?_⟩
+        have : ExpChar (R ⊗[k] K) q := by
+          refine expChar_of_injective_ringHom
+            (f := TensorProduct.includeRight.toRingHom.comp (algebraMap k K)) ?_ q
+          exact (Algebra.TensorProduct.includeRight_injective (algebraMap k R).injective).comp
+            (algebraMap k K).injective
+        rw [add_pow_expChar_pow, pow_add]
+        nth_rw 2 [mul_comm]
+        rw [pow_mul, pow_mul]
+        exact Subring.add_mem _ (Subring.pow_mem _ hx _) (Subring.pow_mem _ hy _)
+      | tmul x y =>
+        obtain ⟨n, a, ha⟩ := IsPurelyInseparable.pow_mem k q y
+        refine ⟨n + 1, by simp, ?_⟩
+        have : (x ^ q ^ (n + 1)) ⊗ₜ[k] (y ^ q ^ (n + 1)) =
+            (x ^ q ^ (n + 1)) ⊗ₜ[k] (1 : K) * (1 : R) ⊗ₜ[k] (y ^ q ^ (n + 1)) := by
+          rw [TensorProduct.tmul_mul_tmul, mul_one, one_mul]
+        rw [TensorProduct.tmul_pow, this]
+        refine Subring.mul_mem _ ⟨x ^ q ^ (n + 1), rfl⟩ ⟨algebraMap k R (a ^ q), ?_⟩
+        rw [pow_add, pow_mul, ← IsScalarTower.algebraMap_apply, TensorProduct.algebraMap_apply,
+          TensorProduct.tmul_one_eq_one_tmul, map_pow, ha, pow_one]
```

Comment:
Perhaps this part should also be its own lemma (`∃ n > 0, x ^ q ^ n ∈ (algebraMap R (R ⊗[k] K)).range` given `ExpChar k q`) and both of them should be somewhere near `IsPurelyInseparable`.